src/som/vmobjects/SClass.java

```diff
@@ -191,17 +199,39 @@ public void initializeStructure(final MixinDefinition mixinDef,
     this.isTransferObject = isTransferObject;
     this.isArray = isArray;
     this.instanceClassGroup = classFactory;
+
+    if (VmSettings.SNAPSHOTS_ENABLED) {
```

Comment:
Could this change go into its own method?

Reply:
i can make an initializeForSnapshotting method